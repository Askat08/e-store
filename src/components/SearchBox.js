import React from "react";

export default function SearchBox({
  onSearchClick,
  onSearchChange,
  searchField,
}) {
  return (
    <div className="input-group col-sm-9 mx-auto">
      <input
        className="form-control py-2 shadow-none"
        type="search"
        value={searchField}
        placeholder="Search..."
        id="example-search-input"
        onChange={onSearchChange}
        onKeyPress={onSearchClick}
      />
      <span className="input-group-append">
        <button
          className="btn btn-outline-info shadow-none"
          type="button"
          onClick={onSearchClick}
        >
          <i className="fa fa-search"></i>
        </button>
      </span>
    </div>
  );
}
