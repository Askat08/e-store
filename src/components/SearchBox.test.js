import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import SearchBox from "./SearchBox";

const simulateInputChange = (wrapper, inputSearch, newValue) => {
  let input = wrapper.find(inputSearch);
  input.simulate("change", {
    target: { value: newValue },
  });
  return wrapper.find(inputSearch);
};

describe("SearchBox component test", () => {
  test("should render app", () => {
    const wrapper = shallow(<SearchBox onChange={simulateInputChange} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test("should render app", () => {
    // const wrapper = shallow(<SearchBox />);
    // const searchInput = simulateInputChange(wrapper, "input", "thor");
    // expect(searchInput.prop("value")).toEqual("");
  });
});
