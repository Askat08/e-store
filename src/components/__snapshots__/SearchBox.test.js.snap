// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox component test should render app 1`] = `
<div
  className="input-group col-sm-9 mx-auto"
>
  <input
    className="form-control py-2 shadow-none"
    id="example-search-input"
    placeholder="Search..."
    type="search"
  />
  <span
    className="input-group-append"
  >
    <button
      className="btn btn-outline-info shadow-none"
      type="button"
    >
      <i
        className="fa fa-search"
      />
    </button>
  </span>
</div>
`;
